mesheryctl/pkg/utils/helpers_test.go

```diff
@@ -0,0 +1,504 @@
+func TestSetFileLocation(t *testing.T) {
+	err := SetFileLocation()
+	if err != nil {
+		t.Errorf("SetFileLocation error = %v", err)
+	}
+}
+
+// func TestNavigateToBrowser(t *testing.T) {
+// 	// opens up a browser window whenever this test runs
+// 	err := NavigateToBrowser("https://www.layer5.io")
+// 	if err != nil {
+// 		t.Errorf("NavigateToBrowser error: %v", err)
+// 	}
+// }
+
+func TestUploadFileWithParams(t *testing.T) {
+	fixtureFileName := "listmanifest.api.response.golden" // any arbitrary fixture file
+	uploadFilePath := filepath.Join(fixturesDir, fixtureFileName)
+
+	test := struct {
+		uri       string
+		params    map[string]string
+		paramName string
+		path      string
+	}{
+		uri:       "https://www.layer5.io",
+		params:    nil,
+		paramName: "meshery",
+		path:      uploadFilePath,
+	}
+
+	// returns *http.Request
+	_, err := UploadFileWithParams(test.uri, test.params, test.paramName, test.path)
+
+	if err != nil {
+		t.Errorf("UploadFileWithParams error = %v", err)
+	}
+}
+
+func TestContentTypeIsHTML(t *testing.T) {
+	tests := []struct {
+		name     string
+		response *http.Response
+		want     bool
+	}{
+		{
+			name: "correct content-type",
+			response: &http.Response{
+				Header: http.Header{
+					"Content-Type": []string{"text/html"},
+				},
+			},
+			want: true,
+		},
+		{
+			name: "empty content-type",
+			response: &http.Response{
+				Header: http.Header{
+					"Content-Type": []string{},
+				},
+			},
+			want: false,
+		},
+		{
+			name: "incorrect content-type",
+			response: &http.Response{
+				Header: http.Header{
+					"Content-Type": []string{"multipart/form-data"},
+				},
+			},
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := ContentTypeIsHTML(tt.response)
+			if got != tt.want {
+				t.Errorf("ContentTypeIsHTML error = %v want = %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestAskForConfirmation(t *testing.T) {
+	tests := []struct {
+		name     string
+		question string
+		input    string
+		want     bool
+	}{
+		{
+			name:     "test with input 'yes'",
+			question: "question?",
+			input:    "yes\n",
+			want:     true,
+		},
+		{
+			name:     "test with input 'no'",
+			question: "question?",
+			input:    "no\n",
+			want:     false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// mocking stdio
+			// see https://stackoverflow.com/a/64518829
+			r, w, err := os.Pipe()
+			if err != nil {
+				t.Fatal(err)
+			}
+			_, err = w.WriteString(tt.input)
+			if err != nil {
+				t.Error(err)
+			}
+			w.Close()
+			stdin := os.Stdin
+			defer func() { os.Stdin = stdin }()
+			os.Stdin = r
+
+			got := AskForConfirmation(tt.question)
+			if got != tt.want {
+				t.Errorf("AskForConfirmation got = %v want = %v", got, tt.want)
+			}
+		})
+	}
+}
+
+// func TestCreateConfigFile(t *testing.T) {
+// 	err := CreateConfigFile()
+// 	if err != nil {
+// 		t.Errorf("CreateConfigFile error = %v", err)
+// 	}
+// }
```

Comment:
I think we can test it out, just update the related variable accordingly.

Reply:
As in change where the config file is being created by modifying the underlying path?

Reply:
Not sure what you mean, I meant to set the variable value from our side to `$HOME/.meshery/config.yaml`

Reply:
The thing is the default path for config file is `$HOME/.meshery/config.yaml`, and it was generating an error in automated tests. Right now, I have relocated this file above `SetFileLocation` function, since `SetFileLocation` is what changed the path to `$HOME/.meshery/config.yaml`.

Let me know what you think of it.

Reply:
yeah let's setup the path ourselves and see if it actually creates the file there.